### PR TITLE
KAZOO-5063 ensure to-did not being empty

### PR DESCRIPTION
**Pull request by jamhed**
Messages with is_empty To-DID field (<<"0">> as example) should not pass validation.

### CODE REVIEW

File: core/kazoo_amqp/src/api/kapi_offnet_resource.erl [hunk 1]
```diff
@@ -187,7 +187,8 @@ req(JObj) -> req(kz_json:to_proplist(JObj)).
 
 -spec req_v(api_terms()) -> boolean().
 req_v(Prop) when is_list(Prop) ->
-    kz_api:validate(Prop, ?OFFNET_RESOURCE_REQ_HEADERS, ?OFFNET_RESOURCE_REQ_VALUES, ?OFFNET_RESOURCE_REQ_TYPES);
+    kz_api:validate(Prop, ?OFFNET_RESOURCE_REQ_HEADERS, ?OFFNET_RESOURCE_REQ_VALUES, ?OFFNET_RESOURCE_REQ_TYPES)
+        andalso not kz_util:is_empty(props:get_value(?KEY_TO_DID, Prop));
```
Comment (lazedo) [inline core/kazoo_amqp/src/api/kapi_offnet_resource.erl]: handle `KEY_TO_DID` values in `OFFNET_RESOURCE_REQ_TYPES`, presently is `,{?KEY_TO_DID, fun is_binary/1}`
resuming, you should try not add these kind of conditions here.

Comment (jamhed) [inline core/kazoo_amqp/src/api/kapi_offnet_resource.erl]: <<"0">> is binary and kz_util:is_empty at the same time, so these checks are not enough. There is no other way to ensure value to be non-empty (except changing kz_json:get_ne_binary behaviour, what i think is not what we want here).

Comment (lazedo) [inline core/kazoo_amqp/src/api/kapi_offnet_resource.erl]: what part did you not understand ?

Comment (fenollp) [inline core/kazoo_amqp/src/api/kapi_offnet_resource.erl]: @jamhed I think @lazedo is saying to add this line over at/around https://github.com/2600hz/kazoo/blob/589a54952a51f35d1f7cf7cd4533c9f3bd13eca7/core/kazoo_amqp/src/api/kapi_offnet_resource.erl#L147

File: core/kazoo_amqp/src/api/kapi_offnet_resource.erl [hunk 2]
```diff
@@ -170,6 +170,11 @@
                                      ]).
 -define(OFFNET_RESOURCE_RESP_TYPES, []).
 
+-spec is_ne_binary(binary()) -> boolean().
+is_ne_binary(V) ->
```
Comment (lazedo) [inline core/kazoo_amqp/src/api/kapi_offnet_resource.erl]: move this to `kz_util`

**Pull request merged**